### ISSUE TITLE
[FEATURE REQ] Add property to disable Kafka related auto-configuration

### ISSUE THREAD

**Issue by ahrytsiuk** [Azure/azure-sdk-for-java#30574]
**Is your feature request related to a problem? Please describe.**
When developing an application that supports multiple Spring Cloud Stream binders:
- `spring-cloud-azure-stream-binder-servicebus`
- `spring-cloud-stream-binder-kafka`
- `spring-cloud-stream-binder-rabbit`

an application fails to connect to on-premise Kafka (which is not deployed in Azure). The root cause - `AzureEventHubsKafkaOAuth2AutoConfiguration` changes user-agent for kafka-client (see class constructor) from `apache-kafka-java` to `apache-kafka-java/az-sp-kafka`. And then when client tries to connect to Kafka I get errors:
```
Received error INVALID_REQUEST from node -1 when making an ApiVersionsRequest with correlation id 9. Disconnecting.
```

**Describe the solution you'd like**
Add `@ConditionalOnProperty` annotation on `AzureEventHubsKafkaOAuth2AutoConfiguration`, so it can be disabled using configuration properties. For example: `spring.cloud.azure.kafka.enabled`.

**Comment by yiliuTo** [comment]
Hi @ahrytsiuk thanks for reaching us, the pr #30605 will meet your request, and we are going to use the property of `spring.cloud.azure.eventhubs.kafka.enabled` to control the auto-configuraion.

**Comment by yiliuTo** [comment]
I will close this issue since it has been resolved in #30605, and you could reopen it for any other questions.
